src/rmp/src/Restructure.cpp

```diff
@@ -524,20 +522,10 @@ void Restructure::writeOptCommands(std::ofstream& script)
 
 void Restructure::setMode(const char* mode_name)
 {
-  if (!strcmp(mode_name, "delay1") || !strcmp(mode_name, "delay"))
+  if (!strcmp(mode_name, "timing"))
     opt_mode_ = Mode::DELAY_1;
-  else if (!strcmp(mode_name, "delay2"))
-    opt_mode_ = Mode::DELAY_2;
-  else if (!strcmp(mode_name, "delay3"))
-    opt_mode_ = Mode::DELAY_3;
-  else if (!strcmp(mode_name, "delay4"))
-    opt_mode_ = Mode::DELAY_4;
-  else if (!strcmp(mode_name, "area1") || !strcmp(mode_name, "area"))
+  else if (!strcmp(mode_name, "area"))
     opt_mode_ = Mode::AREA_1;
-  else if (!strcmp(mode_name, "area2"))
-    opt_mode_ = Mode::AREA_2;
-  else if (!strcmp(mode_name, "area3"))
-    opt_mode_ = Mode::AREA_3;
```

Comment:
Are the other modes obsolete?

Reply:
Externally we are supporting only 2 modes: timing and area
Within timing and area, the submodes, are not visible externally now. Earlier I was thinking to keeping them external as well.